### DISCUSSION TITLE
RAD Files and Mixed Species experiment.

### DISCUSSION THREAD

**Discussion by AndreaMariani-AM** [COMBINE-lab/alevin-fry#119] [Q&A]
Hi all,

First of all thank you for the amazing suite of tools for the whole ecosystem.

Now, i'm analysing a SPLiT-seq experiment as first described by [Rosemberg et al.](https://www.science.org/doi/10.1126/science.aam8999) using Drop-seq but i would really like to fully commit to alevin-fry workflow given that i already use salmon for my normal bulk RNAseq experiments. In doing so, i face 2 main "problems" which i want to discuss here:

1) RAD files. Maybe i've missed it but there seems to be not a huge [documentation](https://static-content.springer.com/esm/art%3A10.1038%2Fs41592-022-01408-3/MediaObjects/41592_2022_1408_MOESM1_ESM.pdf) on them and i don't know if there are in place methods to be able to convert /interact with them. This is the first problem i'm facing, when existing tools require, for example, a BAM file to process. Of course, BAM files and RAD(BUS) files are very different in nature and therefore there's not an easy way to make them coexist, but i was wondering if you guys where thinking about these specific problems and planing some features like [inspect or text](https://bustools.github.io/manual) from BUStools.

2) This second issue somewhat arises from the first one. The main reason i'm still using drop-seq to test/set up the workflow in my lab is that i don't see how i would be able to use the alevin-fry workflow for a mixed species experiment (human, mouse). I'll explain it better. Once nice thing that you can do with [Drop-seq](https://mccarrolllab.org/wp-content/uploads/2016/03/Drop-seqAlignmentCookbookv1.2Jan2016.pdf) is, once you've selected the number of "true" barcodes is your experiment, divide the aligned BAM by specie, quantify the expression of human and mouse transcripts for each barcode (as the aligned reads in the BAM file have been annotated with the associated feature [Gene, exon ...] with the prepended HUMAN_ or MOUSE_ that can be extracted) and get different specie specific statistics (for example, a simple [Mouse,Human transcripts plot](https://www.science.org/doi/10.1126/science.aam8999), fig 1.B). In my case, these analyses are crucial to set the workflow for the lab/unit to be able to implement the method stably (although i know most experiment would be different and wouldn't be mixed species and therefore the "issue" might not be even relevant). If you have any suggestions on this, please feel free to add anything to this as i don't really know how to get this done only with alevin-fry.

I hope i have been clear enough and thanks again.

ps: in the [alevin-fry splitseq tutorial](https://combine-lab.github.io/alevin-fry-tutorials/2022/split-seq/) there's a small error, in the section QUANTIFICATION when explaining the different chemistries v1 and v2. 

"Both are similar, with the only difference being in the position of the third barcode. In v1 it occurs at position 78 and in v2 it occurs at position 86 of the sequenced read. " should be the opposite. in V1 it occurs at 86 and in v2 it occurs at 78, which is consistent with what has been coded in [protocols](https://github.com/COMBINE-lab/salmon/blob/master/include/SingleCellProtocols.hpp)

Thanks,
Andre

**Comment by rob-p** [comment]
Hi @AndreaMariani-AM,

  Thank you for the detailed questions / topics, and for raising these important points for discussion.  Hopefully, I can shed some light on these topics:

> RAD files. Maybe i've missed it but there seems to be not a huge [documentation](https://static-content.springer.com/esm/art%3A10.1038%2Fs41592-022-01408-3/MediaObjects/41592_2022_1408_MOESM1_ESM.pdf) on them and i don't know if there are in place methods to be able to convert /interact with them. This is the first problem i'm facing, when existing tools require, for example, a BAM file to process. Of course, BAM files and RAD(BUS) files are very different in nature and therefore there's not an easy way to make them coexist, but i was wondering if you guys where thinking about these specific problems and planing some features like [inspect or text](https://bustools.github.io/manual) from BUStools.

This is *absolutely* something that we are working on.  The `RAD` format has an internal specification, but you are correct that we have not published it widely yet.  There are several reasons for this, but the main one is that we have several specific use-cases in mind for which we hope to use the `RAD` format, including for storing mapping information for bulk RNA-seq and for general mapping information in the context of metagenomics / pangenomics.  As such, while the format used for single-cell is mostly fixed right now, we'd like to nail down the details of these other use cases before we publish a formal spec.  However, we are planning to do this, as well as to provide some other tools and libraries for general RAD manipulation.  I'll note that `alevin-fry` *does* currently have a `view` command that will print out a plain-text view of a binary RAD file. It can be useful to inspect the mapping information if you want.


> This second issue somewhat arises from the first one. The main reason i'm still using drop-seq to test/set up the workflow in my lab is that i don't see how i would be able to use the alevin-fry workflow for a mixed species experiment (human, mouse). I'll explain it better. Once nice thing that you can do with [Drop-seq](https://mccarrolllab.org/wp-content/uploads/2016/03/Drop-seqAlignmentCookbookv1.2Jan2016.pdf) is, once you've selected the number of "true" barcodes is your experiment, divide the aligned BAM by specie, quantify the expression of human and mouse transcripts for each barcode (as the aligned reads in the BAM file have been annotated with the associated feature [Gene, exon ...] with the prepended HUMAN_ or MOUSE_ that can be extracted) and get different specie specific statistics (for example, a simple [Mouse,Human transcripts plot](https://www.science.org/doi/10.1126/science.aam8999), fig 1.B). In my case, these analyses are crucial to set the workflow for the lab/unit to be able to implement the method stably (although i know most experiment would be different and wouldn't be mixed species and therefore the "issue" might not be even relevant). If you have any suggestions on this, please feel free to add anything to this as i don't really know how to get this done only with alevin-fry.

This is already supported, and we've processed several such "barnyard" experiments with `alevin-fry`.  Essentially, all you have to do is to create an index with the transcriptomes of both organisms.  Subsequently, quantification will take place against the entire indexed (hybrid) transcriptome.  The resulting count matrix will be produced and then counts can be separated post-hoc based on the gene names.  If you have any specific trouble building such a pipeline for your use-case, let us know and we'd be happy to help troubleshoot.  Nonetheless, this should already work without too much hassle!

> ps: in the [alevin-fry splitseq tutorial](https://combine-lab.github.io/alevin-fry-tutorials/2022/split-seq/) there's a small error, in the section QUANTIFICATION when explaining the different chemistries v1 and v2.

Thanks for this! We'll go back and take a look and try and correct anything that's wrong or unclear.

Best,
Rob

**Comment by AndreaMariani-AM** [comment]
Hi @rob-p  ,

Thanks for the answer. I absolutely got your point, it wasn't a critique, more a way to find out some additional info on RAD files. 

> This is _absolutely_ something that we are working on. The `RAD` format has an internal specification, but you are correct that we have not published it widely yet. There are several reasons for this, but the main one is that we have several specific use-cases in mind for which we hope to use the `RAD` format, including for storing mapping information for bulk RNA-seq and for general mapping information in the context of metagenomics / pangenomics. As such, while the format used for single-cell is mostly fixed right now, we'd like to nail down the details of these other use cases before we publish a formal spec. However, we are planning to do this, as well as to provide some other tools and libraries for general RAD manipulation. 
>

 You are absolutely right, i'm sorry i've overlooked it. 
> I'll note that `alevin-fry` _does_ currently have a `view` command that will print out a plain-text view of a binary RAD file. It can be useful to inspect the mapping information if you want.
> 

 True, i've done the combined transcriptome a while ago and completely forgotten they had been prepended with a specie specific tag. Thus i can  divide them based on gene names and count UMI across barcodes per specie.
> This is already supported, and we've processed several such "barnyard" experiments with `alevin-fry`. Essentially, all you have to do is to create an index with the transcriptomes of both organisms. Subsequently, quantification will take place against the entire indexed (hybrid) transcriptome. The resulting count matrix will be produced and then counts can be separated post-hoc based on the gene names. If you have any specific trouble building such a pipeline for your use-case, let us know and we'd be happy to help troubleshoot. Nonetheless, this should already work without too much hassle!
> 
 Thank you again for the effort and preciseness in replying.
 Best Andrea
